File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:

        longest = 0
        seen = set()
        left = 0
        for right in range(len(s)):
            c_char = s[right]
            while c_char in seen:
                seen.remove(s[left])
                left += 1
            seen.add(c_char)
            longest = max(longest, len(seen))
        

        return longest



        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        
        HashSet<char> seen = new HashSet<char>();
        int longest = 0;
        int left = 0;
        
        for (int right = 0; right < s.Length; right++) {
            if (!seen.Contains(s[right])) {
                seen.Add(s[right]);
                longest = Math.Max(longest, right - left + 1);
                continue;
            }
            seen.Remove(s[left]);
            right -= 1;
            left += 1;
        }
        
        
        
        
        
        
        return longest;
        
        
    }
}